libraries/AP_Math/scurves.cpp

```diff
@@ -0,0 +1,645 @@
+    pos += _delta_unit_3 * (segment[11].start_pos - scurve_P1);
+    vel += _delta_unit_3 * scurve_V1;
+    accel += _delta_unit_3 * (-scurve_A1);
+
+    finish = _t > (time_mid + 2.0 * time_start);
+    finish = _t > (time_mid + 2.0 * time_start);
+
+    vel *= MIN(time_scale * 1.1f, 1.0f);
+    accel *= time_scale;
+    return finish;
+}
+
+// debugging messages
+void scurves::debug()
+{
+    hal.console->printf("\n");
+    hal.console->printf("num_segs:%u, type:%4.2f, _t:%4.2f, otj:%4.2f, jerk_max:%4.2f, accel_max:%4.2f, vel_max:%4.2f\n",
+                        (unsigned)num_segs, (double)segtype, (double)_t, (double)otj, (double)jerk_max, (double)accel_max, (double)vel_max);
+    hal.console->printf("T, Jt, J, A, V, P \n");
+    for (uint8_t i = 0; i < num_segs; i++) {
+        hal.console->printf("i:%u, T:%4.2f, Jtype:%4.2f, J:%4.2f, A:%4.2f, V: %4.2f, P: %4.2f\n",
+                            (unsigned)i, (double)segment[i].start_time, (double)segment[i].jtype, (double)segment[i].jerk_ref,
+                            (double)segment[i].start_accel, (double)segment[i].start_vel, (double)segment[i].start_pos);
+    }
+    hal.console->printf("_track x:%4.2f, y:%4.2f, z:%4.2f\n", (double)_track.x, (double)_track.y, (double)_track.z);
+    hal.console->printf("_delta_unit_1 x:%4.2f, y:%4.2f, z:%4.2f\n", (double)_delta_unit_1.x, (double)_delta_unit_1.y, (double)_delta_unit_1.z);
+    hal.console->printf("_delta_unit_2 x:%4.2f, y:%4.2f, z:%4.2f\n", (double)_delta_unit_2.x, (double)_delta_unit_2.y, (double)_delta_unit_2.z);
+    hal.console->printf("_delta_unit_2 x:%4.2f, y:%4.2f, z:%4.2f\n", (double)_delta_unit_3.x, (double)_delta_unit_3.y, (double)_delta_unit_3.z);
+    hal.console->printf("\n");
+}
+
+// straight segment implementations of pos_end, time_end, time_to_end and braking
+float scurves::pos_end_straight() const
+{
+    return segment[num_segs - 1].start_pos;
+}
+
+float scurves::time_end_straight() const
+{
+    return segment[num_segs - 1].start_time;
+}
+
+float scurves::get_time_remaining_straight() const
+{
+    return segment[num_segs - 1].start_time - _t;
+}
+
+bool scurves::braking_straight() const
+{
+    return _t >= segment[8].start_time;
+}
+
+// spline segment implementations of pos_end, time_end, time_to_end and braking
+float scurves::pos_end_spline() const
+{
+    return segment[num_segs - 1].start_pos + segment[11].start_pos;
+}
+
+float scurves::time_end_spline() const
+{
+    return segment[num_segs - 1].start_time - segment[11].start_time + segment[7].start_time * 2.0;
+}
+
+float scurves::get_time_remaining_spline() const
+{
+    return time_end_spline() - _t;
+}
+
+bool scurves::braking_spline() const
+{
+    return _t > segment[num_segs - 1].start_time - segment[11].start_time + segment[7].start_time;
+}
+
+// generate constant jerk time segment
+void scurves::add_segment_const_jerk(float tin, float J0)
+{
+    enum jtype_t Jtype = jtype_t::CONSTANT;
```

Comment:
`const`

Reply:
I've removed this line and the constant now appears directly as an argument to the add_segment method.